src/entryproc.cpp

```diff
@@ -47,9 +47,12 @@ bool EntryProcessor::process(const bool have_proc) {
             m_safe_path = std::string("/proc/self/fd/") + std::to_string(m_fd.get());
         } else {
             // fall back to plain path-access for read-only operation. (this much is fine)
-            // we only report errors below, m_apply_changes is set to false in this case.
+            // unless CHKSTAT_FORCE_INSECURE_MODE_IF_NO_PROC is set we only report errors below, m_apply_changes is set to false in this case.
+            // if CHKSTAT_FORCE_INSECURE_MODE_IF_NO_PROC is set we operate insecurely, see bsc#1219736
             m_safe_path = m_entry.file;
-            assert(!m_apply_changes);
+            if (secure_getenv("CHKSTAT_FORCE_INSECURE_MODE_IF_NO_PROC") == nullptr) {
```

Comment:
Can you avoid the two redundant `secure_getenv()` calls? You could add a member variable `const m_allow_no_proc` in the Chkstat main class. The assertion can be placed into `Chkstat` itself in the beginning of `processEntries()`.

Reply:
A single definition of the environment variable name would also be helpful, a `constexpr std::string_view ENVVAR_ALLOW_NO_PROC{"CHKSTAT_..."}` somewhere.

Reply:
done